sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -134,37 +144,168 @@ export class IER12ProcessingService {
    * @param pendingAssessment pending assessment of institutions.
    * @returns IER 12 record.
    */
-  private createIER12Record(pendingAssessment: StudentAssessment): IER12Record {
+  private async createIER12Record(
+    pendingAssessment: StudentAssessment,
+  ): Promise<IER12Record[]> {
     const application = pendingAssessment.application;
     const student = application.student;
     const user = student.user;
     const sinValidation = student.sinValidation;
     const offering = pendingAssessment.offering;
     const educationProgram = offering.educationProgram;
-    return {
-      assessmentId: pendingAssessment.id,
-      applicationNumber: application.applicationNumber,
-      sin: sinValidation.sin,
-      studentLastName: user.lastName,
-      studentGivenName: user.firstName,
-      birthDate: student.birthDate,
-      programName: educationProgram.name,
-      programDescription: educationProgram.description,
-      credentialType: educationProgram.credentialType,
-      cipCode: parseFloat(educationProgram.cipCode) * 10000,
-      nocCode: educationProgram.nocCode,
-      sabcCode: educationProgram.sabcCode,
-      institutionProgramCode: educationProgram.institutionProgramCode,
-      programLength: offering.yearOfStudy,
-      studyStartDate: offering.studyStartDate,
-      studyEndDate: offering.studyEndDate,
-      tuitionFees: offering.actualTuitionCosts,
-      programRelatedCosts: offering.programRelatedCosts,
-      mandatoryFees: offering.mandatoryFees,
-      exceptionExpenses: offering.exceptionalExpenses,
-      totalFundedWeeks: offering.studyBreaks.totalFundedWeeks,
-      disbursementSchedules: pendingAssessment.disbursementSchedules,
+    const address = student.contactInfo.address;
+    const applicationProgramYear = application.programYear;
+    const [scholasticStanding] = application.studentScholasticStandings;
+    const assessmentAwards = this.getAssessmentAwards(
+      pendingAssessment.disbursementSchedules,
+    );
+    const disbursementSchedules = pendingAssessment.disbursementSchedules;
+    const addressInfo: IERAddressInfo = {
+      addressLine1: address.addressLine1,
+      addressLine2: address.addressLine2,
+      city: address.city,
+      provinceState: address.provinceState,
+      postalCode: address.postalCode,
     };
+    const hasRestriction = this.checkActiveRestriction(
+      student.studentRestrictions,
+    );
+    const hasProvincialDefaultRestriction = !hasRestriction
+      ? false
+      : this.checkActiveRestriction(student.studentRestrictions, {
+          restrictionCode: "B2",
+        });
+    const hasPartner =
```

Comment:
The relationship status is saved to the application table, is there a reason to not read it from there?

Reply:
You the relationship status is application dynamic data in applications table ? - If so we decided on analysis to read the variables which impact calculations from workflow right? let me know if I am missing something here.

Maybe the e-Cert should start reading from workflowData.studentData.relationshipStatus also? - Yes I feel that is must happen.

Reply:
As discussed, added the question to the story.

![image](https://github.com/bcgov/SIMS/assets/54600590/d6f40f3b-4eb7-4b57-b96b-66cb48510889)